app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,59 @@
+<div class="container recipe-details">
+  <h2 class="text-center mb-4" style="margin-right:30%;"><%= @recipe.name %></h2>
+  <div class="row recipe-details-box1">
+    <div class="col-md-6">
+      <p><strong>Preparation Time:</strong> <%= @recipe.preparation_time %> minutes</p>
+    </div>
+
+    <div class="col-md-6 text-right">
+      <%= form_for @recipe, url: recipe_path(@recipe), method: :patch do |f| %>
+        <div class="form-group">
+           <% if @recipe.public == 1 %>
+            <p>Public: Yes</p>
+            <% else %>
+            <p>Public: No</p>
+            <% end %>
+          <%= f.submit 'Toggle', class: 'btn btn-primary' %>
+        </div>
+      <% end %>
+    </div>
+</div>
+
+  <p><strong>Cooking Time:</strong> <%= @recipe.cooking_time %> minutes</p>
+  <p><strong>Description:</strong> <%= @recipe.description %></p>
+  
+  <div class="row recipe-details-btns mt-4 mb-4">
+    <div class="col-md-6">
+      <%= link_to "Generate Shopping List", general_shopping_list_path, class: "btn btn-success btn-block" %>
+    </div>
+    <div class="col-md-6">
+      <% if current_user == @recipe.user %>
+        <%= link_to "Add Ingredient", new_recipe_recipe_food_path(recipe_id: @recipe.id), class: 'btn btn-primary btn-block' %>
+      <% end %>
+    </div>
+  </div>
+
+  <table class="table mt-4">
+    <thead>
+      <tr>
+        <th>Food</th>
+        <th>Quantity</th>
+        <th>Value</th>
+        <th>Actions</th>
```

Comment:
- [ ] On The recipe show page we should be able to generate a shopping list. Kindly fix this issue.


Videocast
[cinnamon-2023-12-22T172824+0200.webm](https://github.com/zunairkhan811/food-recipe/assets/81932478/4da18548-899a-4f41-bfe5-c135e5641577)

Reply:
@Munsa1 Thanks for your suggestions. So that you know – the changes have been made as you said.